src/main/java/org/apache/cassandra/sidecar/cluster/ConsistencyVerifiers.java

```diff
@@ -0,0 +1,356 @@
+                });
+            });
+
+            UnaryOperator<String> dcClassifier = s -> {
+                String dc = dcByInstance.get(s);
+                return dc != null ? dc : UNKNOWN_DC;
+            };
+            InstanceSetByDc passedByDc = groupByDc(succeeded, dcClassifier);
+            validateNoneFromUnknownDc(passedByDc, "passed");
+            InstanceSetByDc failedByDc = groupByDc(failed, dcClassifier);
+            validateNoneFromUnknownDc(failedByDc, "failed");
+
+            // check whether passed instance can satisfy quorum in all DCs
+            boolean allSatisfied = true;
+            boolean anyFailed = false;
+            for (String dcName : all.keySet())
+            {
+                // if any of the datacenter cannot satisfy quorum locally
+                if (!geQuorum(passedByDc.get(dcName).size(), all.get(dcName).size()))
+                {
+                    allSatisfied = false;
+                }
+
+                // if any of the datacenter has failed quorum locally
+                if (geQuorum(failedByDc.get(dcName).size(), all.get(dcName).size()))
+                {
+                    anyFailed = true;
+                }
+            }
+            if (allSatisfied)
+            {
+                return Result.SATISFIED;
+            }
+
+            if (anyFailed)
+            {
+                return Result.FAILED;
+            }
+
+            return Result.PENDING;
+        }
+    }
+
+    /**
+     * Verifier for consistency level QUORUM
+     * It concludes result
+     * - SATISFIED: quorum instances succeed
+     * - FAILED: quorum instances fail
+     * - PENDING: default
+     */
+    public static class ForQuorum extends BaseVerifier
+    {
+        public static final ForQuorum INSTANCE = new ForQuorum();
+
+        @Override
+        public Result verify(Set<String> succeeded, Set<String> failed, InstanceSetByDc all)
+        {
+            if (geQuorum(succeeded.size(), sum(all)))
+            {
+                return Result.SATISFIED;
+            }
+
+            if (geQuorum(failed.size(), sum(all)))
+            {
+                return Result.FAILED;
+            }
+
+            return Result.PENDING;
+        }
+    }
+
+    /**
+     * Verifier for consistency level ALL
+     * It concludes result
+     * - SATISFIED: all instances succeed
+     * - FAILED: one instance fails
+     * - PENDING: default
+     */
+    public static class ForAll extends BaseVerifier
+    {
+        public static final ForAll INSTANCE = new ForAll();
+
+        @Override
+        public Result verify(Set<String> succeeded, Set<String> failed, InstanceSetByDc all)
+        {
+            if (succeeded.size() == sum(all))
+            {
+                return Result.SATISFIED;
+            }
+
+            if (!failed.isEmpty())
+            {
+                return Result.FAILED;
+            }
+
+            return Result.PENDING;
```

Comment:
Ideally the failed and succeeded sets are mutually exclusive, but I"m thinking of the case where something is wrong from the caller, we can return an accurate Result by doing this:

```suggestion
            Result result = Result.PENDING;
            if (succeeded.size() == sum(all))
            {
                result = Result.SATISFIED;
            }

            if (!failed.isEmpty())
            {
                result = Result.FAILED;
            }

            return result;
```